### PR TITLE
Add build target for FreeBSD-amd64

### PR DESCRIPTION
**Pull request by tehpeh**
Changes:

- "set -o pipefail" isn't available yet in sh for all
versions of FreeBSD, so use bash explicitly.
-  use "printf" over "echo" as it behaves more consistently
across platforms, ie. with newlines.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,16 +53,20 @@ rubygem/build/darwin-amd64/ejson: build/bin/darwin-amd64
...
+	mkdir -p $(@D)
+	cp -a "$<" "$@"
+
 rubygem/build/linux-amd64/ejson: build/bin/linux-amd64
 	mkdir -p $(@D)
 	cp -a "$<" "$@"
 
 cmd/$(NAME)/version.go: VERSION
-	echo 'package main\n\nconst VERSION string = "$(VERSION)"' > $@
+	printf 'package main\n\nconst VERSION string = "$(VERSION)"' > $@
```
Comment (lavoiesl) [inline Makefile]: Isn't this missing a newline?
```suggestion
	printf 'package main\n\nconst VERSION string = "$(VERSION)"\n' > $@
```

File: Makefile [hunk 2]
```diff
@@ -50,16 +53,20 @@ rubygem/build/darwin-amd64/ejson: build/bin/darwin-amd64
...
 	cp -a "$<" "$@"
 
 cmd/$(NAME)/version.go: VERSION
-	echo 'package main\n\nconst VERSION string = "$(VERSION)"' > $@
+	printf 'package main\n\nconst VERSION string = "$(VERSION)"\n' > $@
 
 rubygem/lib/$(NAME)/version.rb: VERSION
 	mkdir -p $(@D)
-	echo 'module $(RUBY_MODULE)\n  VERSION = "$(VERSION)"\nend' > $@
+	printf 'module $(RUBY_MODULE)\n  VERSION = "$(VERSION)"\nend' > $@
```
Comment (lavoiesl) [inline Makefile]: This one is also missing a newline

**Comment by tehpeh** [comment]
@lavoiesl does this PR need anything else?

File: Makefile [hunk 3]
```diff
@@ -14,29 +14,32 @@ BUNDLE_EXEC=bundle exec
...
-binaries: build/bin/linux-amd64 build/bin/darwin-amd64
+binaries: build/bin/linux-amd64 build/bin/darwin-amd64 build/bin/freebsd-amd64
 gem: $(GEM)
 deb: $(DEB)
 man: $(MANFILES)
 
 build/man/%.gz: man/%.ronn
 	mkdir -p "$(@D)"
-	set -euo pipefail ; $(BUNDLE_EXEC) ronn -r --pipe "$<" | gzip > "$@" || (rm -f "$<" ; false)
+	bash -c 'set -euo pipefail ; $(BUNDLE_EXEC) ronn -r --pipe "$<" | gzip > "$@" || (rm -f "$<" ; false)'
```
Comment (burke) [inline Makefile]: Can you try changing this line back but adding `SHELL=/bin/bash` near the top of the Makefile?

File: Makefile [hunk 4]
```diff
@@ -50,16 +53,20 @@ rubygem/build/darwin-amd64/ejson: build/bin/darwin-amd64
...
+	mkdir -p $(@D)
+	cp -a "$<" "$@"
+
 rubygem/build/linux-amd64/ejson: build/bin/linux-amd64
 	mkdir -p $(@D)
 	cp -a "$<" "$@"
 
 cmd/$(NAME)/version.go: VERSION
-	echo 'package main\n\nconst VERSION string = "$(VERSION)"' > $@
+	printf 'package main\n\nconst VERSION string = "$(VERSION)"\n' > $@
```
Comment (burke) [inline Makefile]: We should probably be doing `print "%s" "..."` instead, yeah?

File: Makefile (hunk 3 above)
Comment (tehpeh) [reply Makefile]: Sounds good to me

**Comment by burke** [comment]
LGTM, thanks for the contribution!

**Pull request merged**